.github/workflows/ci.yml

```diff
@@ -10,13 +10,52 @@ permissions:
   contents: read
 
 jobs:
-  ci:
-    runs-on: ubuntu-latest
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - name: std (windows)
+            os: windows-2022
+            target: x86_64-pc-windows-msvc
+            args: ""
+            run_tests: true
+          - name: std (macos)
+            os: macos-latest
+            target: x86_64-apple-darwin
+            args: ""
+            run_tests: true
+          - name: std (linux)
+            os: ubuntu-latest
+            target: x86_64-unknown-linux-gnu
+            args: ""
+            run_tests: true
+          - name: no_std (thumbv7em)
+            os: ubuntu-latest
+            target: thumbv7em-none-eabi
+            args: "--no-default-features"
+            run_tests: false
+          - name: no_std (riscv32 + portable-atomic)
+            os: ubuntu-latest
+            target: riscv32imc-unknown-none-elf
+            args: "--no-default-features --features portable-atomic,portable-atomic/unsafe-assume-single-core"
+            run_tests: false
+
+    name: ${{ matrix.name }}
     steps:
       - uses: actions/checkout@v5
+
       - uses: dtolnay/rust-toolchain@stable
-      - run: cargo build
-      - run: cargo test
+        with:
+          target: ${{ matrix.target }}
+
+      - name: Build
+        run: cargo build --target ${{ matrix.target }} ${{ matrix.args }}
+
+      - name: Test
+        if: matrix.target != 'thumbv7em-none-eabi' && matrix.target != 'riscv32imc-unknown-none-elf'
```

Comment:
I think you meant to use the `run_tests` variable here, right?

Reply:
I did, yes. And in my testing I had it that way.
I'll just fix that up, nice catch